device/core/src/module_client.ts

```diff
@@ -349,6 +379,14 @@ export class ModuleClient extends InternalClient {
     }
   }
 
+  static fromEnvironment(transportCtor: any, callback?: Callback<ModuleClient>): Promise<ModuleClient> | void {
+    if (callback) {
+      return this._fromEnvironment(transportCtor, callback);
+    }
+
+    return callbackToPromise((_callback) => this._fromEnvironmentEdge(transportCtor, _callback));
```

Comment:
This seems like an unnecessary method because callbackToPromise() should work with a user provided callback, right?

Reply:
Done.

---
In reply to: [218892674](https://github.com/Azure/azure-iot-sdk-node/pull/372#discussion_r218892674) [](ancestors = 218892674)